hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java

```diff
@@ -290,16 +321,27 @@ public int getDepth() {
         if (ApiErrorUtils.hasRequestErrors(errorMessages)) {
             return ApiErrorUtils.badRequest(errorMessages);
         } else {
-            switch (command.getCommand()) {
-                case START:
-                    protocolAdapterManager.start(adapterId);
-                    break;
-                case STOP:
-                    protocolAdapterManager.stop(adapterId);
-                    break;
-                case RESTART:
-                    protocolAdapterManager.stop(adapterId).thenRun(() -> protocolAdapterManager.start(adapterId));
-                    break;
+            try {
+                switch (command.getCommand()) {
+                    case START:
+                        protocolAdapterManager.start(adapterId).get();
+                        break;
+                    case STOP:
+                        protocolAdapterManager.stop(adapterId).get();
+                        break;
+                    case RESTART:
+                        protocolAdapterManager.stop(adapterId)
+                                .thenRun(() -> protocolAdapterManager.start(adapterId))
```

Comment:
What happens if the stop never ends?

Reply:
I added timeouts to all adapter interactions. This way we should be safe.